staging/src/k8s.io/sample-controller/main.go

```diff
@@ -58,8 +58,23 @@ func main() {
 		klog.Fatalf("Error building example clientset: %s", err.Error())
 	}
 
-	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, time.Second*30)
-	exampleInformerFactory := informers.NewSharedInformerFactory(exampleClient, time.Second*30)
+	// Informers trigger events on resource changes, which typically queues
+	// a reconciliation of that resource.  The resync period acts as a
+	// safety net to protect against dropped events, and the informer will
+	// signal a "nop" event to trigger a reconciliation of every resource
+	// on this interval.  Small values here have an assortment of problems:
+	//  1. It can mask unhandled events.  If an event is not handled then
```

Comment:
So part of the reason historically for this is that this is a feature while iterating on a controller.  Most controllers that will ever be written are small, have a small working set, and may never get to the point of needing more.  Brute forcing sync is an acceptable solution for that problem domain.

While I’m not opposed to increasing the defaults, its probably also important to communicate the problems this comment describes by changing the parameter name to something that communicates the nature better, forcing someone to set a default, or changing the signature of the higher level code to give this context.  I’m worried that this trades one problem for another, while we want to force controller authors up front to ask “what am I doing when I set resyncninternal”.

10 hours is dangerously high from the perspective of “you won’t catch this problem until 10 hours after you’ve deployed it” as well.

Reply:
As suggested on Twitter, how about 10m?  Controller runtime should probably not be at 10h either, 10m would be a nice compromise.

Reply:
10h might as well be never--which is how I think many controllers ought to run :)

10m is a decent default.

As much as I hate flags, I think there should probably be a flag, to allow and/or encourage people to run with different settings in test vs prod.

Reply:
I have no skin in this game.  Paging @pwittrock :)